src/GafferScene/RenderController.cpp

```diff
@@ -1436,15 +1436,15 @@ void RenderController::setManifestRequired( bool manifestRequired )
 
 	m_manifestRequired = manifestRequired;
 
-	// We apply the manifest update immediately. This allows a client who calls
-	// setManifestRequired( true ) to assume that subsequent calls to renderManifest()
-	// will not return nullptr.
-	if( !manifestRequired )
-	{
-		m_renderManifest.reset();
-	}
-	else
+	// When enabling manifestRequired, we apply the manifest update immediately.
+	// This kind of feels like it is making things more complicated in exchange for less
+	// consistent behaviour ( turning off manifestRequired applies on the next update,
+	// not immediately ) - but the benefit of this special case immediate update is that
+	// it allows a client who calls setManifestRequired( true ) to assume that subsequent
+	// calls to renderManifest() will not return nullptr.
+	if( manifestRequired && !m_renderManifest )
```

Comment:
We could be a bit more selective about when we call `dirtyGlobals()` and `requestUpdate()` - we're calling them unnecessarily in some circumstances. I think the easiest way might be to add this at L1438 :

```
if( m_manifestRequired == (bool)m_manifest )
{
    return;
}
```